usecase/src/test/java/org/mobilitydata/gtfsvalidator/usecase/ProcessParsedShapePointTest.java

```diff
@@ -0,0 +1,195 @@
+        when(mockParsedShape.getEntityId()).thenReturn("shape id");
+        final ShapePoint mockShapePoint = mock(ShapePoint.class);
+
+        //noinspection rawtypes to avoid lind
+        final EntityBuildResult mockGenericObject = mock(EntityBuildResult.class);
+        when(mockGenericObject.isSuccess()).thenReturn(true);
+        when(mockGenericObject.getData()).thenReturn(mockShapePoint);
+
+        when(mockShapePoint.getShapeId()).thenReturn(SHAPE_ID);
+        //noinspection unchecked to avoid lind
+        when(mockBuilder.build()).thenReturn(mockGenericObject);
+        when(mockGtfsDataRepo.addShapePoint(mockShapePoint)).thenReturn(null);
+
+        final ProcessParsedShapePoint underTest = new ProcessParsedShapePoint(mockResultRepo, mockGtfsDataRepo, mockBuilder);
+
+        when(mockParsedShape.get(ArgumentMatchers.eq(SHAPE_ID))).thenReturn("shape id");
+        when(mockParsedShape.get(ArgumentMatchers.eq(SHAPE_PT_LAT))).thenReturn(0f);
+        when(mockParsedShape.get(ArgumentMatchers.eq(SHAPE_PT_LON))).thenReturn(0f);
+        when(mockParsedShape.get(ArgumentMatchers.eq(SHAPE_PT_SEQUENCE))).thenReturn(1);
+        when(mockParsedShape.get(ArgumentMatchers.eq(SHAPE_DIST_TRAVELED))).thenReturn(100.0f);
+
+        underTest.execute(mockParsedShape);
+
+        verify(mockParsedShape, times(1)).get(SHAPE_ID);
+        verify(mockParsedShape, times(1)).get(SHAPE_PT_LAT);
+        verify(mockParsedShape, times(1)).get(SHAPE_PT_LON);
+        verify(mockParsedShape, times(1)).get(SHAPE_PT_SEQUENCE);
+        verify(mockParsedShape, times(1)).get(SHAPE_DIST_TRAVELED);
+        //noinspection ResultOfMethodCallIgnored to avoid lind
+        verify(mockParsedShape, times(1)).getEntityId();
+
+        verify(mockGtfsDataRepo, times(1)).addShapePoint(ArgumentMatchers.eq(mockShapePoint));
+
+
+        verify(mockBuilder, times(1)).shapeId(ArgumentMatchers.eq("shape id"));
+        verify(mockBuilder, times(1)).shapePtLat(ArgumentMatchers.eq(0f));
+        verify(mockBuilder, times(1)).shapePtLon(ArgumentMatchers.eq(0f));
+        verify(mockBuilder, times(1)).shapePtSequence(ArgumentMatchers.eq(1));
+        verify(mockBuilder, times(1)).shapeDistTraveled(ArgumentMatchers.eq(100.f));
+        verify(mockBuilder, times(1)).build();
+
+        verify(mockGenericObject, times(1)).isSuccess();
+        //noinspection ResultOfMethodCallIgnored to avoid lind
+        verify(mockGenericObject, times(1)).getData();
+
+        final ArgumentCaptor<DuplicatedEntityNotice> captor = ArgumentCaptor.forClass(DuplicatedEntityNotice.class);
+
+        verify(mockResultRepo, times(1)).addNotice(captor.capture());
+
+        final List<DuplicatedEntityNotice> noticeList = captor.getAllValues();
+
+        assertEquals("shapes.txt", noticeList.get(0).getFilename());
+        assertEquals(SHAPE_ID, noticeList.get(0).getNoticeSpecific(Notice.KEY_FIELD_NAME));
+        assertEquals("shape id", noticeList.get(0).getEntityId());
+
```

Comment:
Works for the two previous tests, but not in this one since there is an interaction with this mock.

See https://github.com/MobilityData/gtfs-validator/blob/59cebcf5b878ee41d21274e7896516039851c640/usecase/src/test/java/org/mobilitydata/gtfsvalidator/usecase/ProcessParsedShapePointTest.java#L168